tests/onnxruntime/test_optimization.py

```diff
@@ -100,6 +101,23 @@ def test_optimization_details(self):
             self.assertEqual(len(sorted_operators_difference), 0)
             gc.collect()
 
+    def test_optimization_fp16(self):
+        model_name = "hf-internal-testing/tiny-random-distilbert"
+        optimization_config = OptimizationConfig(optimization_level=0, fp16=True)
+        with tempfile.TemporaryDirectory() as tmp_dir:
+            output_dir = Path(tmp_dir)
+            model_path = output_dir.joinpath("model.onnx")
+            optimized_model_path = output_dir.joinpath("model-optimized.onnx")
+            optimizer = ORTOptimizer.from_pretrained(model_name, feature="sequence-classification")
+            optimizer.export(
+                onnx_model_path=model_path,
+                onnx_optimized_model_output_path=optimized_model_path,
+                optimization_config=optimization_config,
+            )
+            model = onnx.load(optimized_model_path.as_posix())
+            for w in model.graph.initializer:
+                self.assertNotEqual(w.data_type, onnx.onnx_pb.TensorProto.FLOAT)
```

Comment:
Why not `self.assertEqual(w.data_type, onnx.onnx_pb.TensorProto.FLOAT16)`?

Reply:
Since there are constants and other int based weights and values

Reply:
for example the inputs are int64

Reply:
Ah okay it makes sense, thanks :)